### PR TITLE
Token config

### PR DESCRIPTION
**Pull request by gititGoro**
The dev-env server now has an API end point "get-token-configs" which returns an array of objects of type
```
interface ITokenConfig {
  displayName: string,
  pyroDisplayName: string,
  address: string,
  pyroV2Address: string
  pyroV3Address:string
}
```
If the baseToken does not have corresponding pyrotokens then the pyroDisplayName is empty and the pyroV2 and pyroV3 addresses are the Ethereum zero address, 0x0000...

No null or undefined values returned.
The tokenConfig array can be an empty array and the api will produce a message stating no tokens were found.

The commit message update provides a bit more info on the necessary deployment script changes for this API update to occur.

### CODE REVIEW

File: dev-env-app/server/src/api/get-deployment-addresses.ts [hunk 1]
```diff
@@ -6,9 +6,9 @@ export default function (fastify: BehodlerDevEnvFastifyInstance, _, done): void
 
   function getDeploymentAddresses() {
     if (fastify.behodlerDevEnv?.active) {
-      return fastify.behodlerDevEnv?.deployedAddresses
+      return fastify.behodlerDevEnv?.set
         ? fastify?.createInfoResponse?.('deployment addresses fetched', {
-            contracts: fastify.behodlerDevEnv?.deployedAddresses,
+            contracts: fastify.behodlerDevEnv?.set.protocol,
           })
         : fastify?.createInfoResponse?.('no deployment addresses found')
```
Comment (khazra) [inline dev-env-app/server/src/api/get-deployment-addresses.ts]: It's probably a bit safer to do
```js
return  fastify.behodlerDevEnv?.set?.protocol
  ? fastify?.createInfoResponse?.('deployment addresses fetched', {
      contracts: fastify.behodlerDevEnv?.set.protocol,
    })
  : fastify?.createInfoResponse?.('no deployment addresses found')
```
Unless the existing `fastify.behodlerDevEnv?.set` assure the `fastify.behodlerDevEnv?.set.protocol` is defined, I'm not sure if this is always the case. I mean, it should be, from the way you coded it, but I'm not 100% sure ;)

File: dev-env-app/server/src/api/get-token-configs.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  fastify.get('/get-token-configs', getTokenConfigs)
+  done()
+
+  function getTokenConfigs() {
+    if (fastify.behodlerDevEnv?.active) {
+      return (fastify.behodlerDevEnv?.set && fastify.behodlerDevEnv?.set.tokens.length > 0)
+        ? fastify?.createInfoResponse?.('tokens found', {
+          contracts: fastify.behodlerDevEnv?.set.tokens,
+        })
+        : fastify?.createInfoResponse?.('no tokens found')
```
Comment (khazra) [inline dev-env-app/server/src/api/get-token-configs.ts]: For a similar reason as in my previous comment, I would change the condition to
```js
(fastify.behodlerDevEnv?.set?.tokens?.length)
```

File: scripts/networks/deploymentFunctions.ts [hunk 3]
```diff
@@ -1833,3 +1835,45 @@ export const prechecks: IDeploymentFunction = async function (params: IDeploymen
   }
   return {}
 }
+
+export const extractTokenConfig = async function (params: IDeploymentParams): Promise<ITokenConfig[]> {
+  const getContract = await getContractFromSection(params.existing)
+  const LR_old = await getContract<Types.LiquidityReceiverV1>(Sections.LiquidityReceiverOld, "LiquidityReceiverV1");
+  const LR_new = await getContract<Types.LiquidityReceiver>(Sections.LiquidityReceiverNew, "LiquidityReceiver");
```
Comment (khazra) [inline scripts/networks/deploymentFunctions.ts:1842]: Just for consideration: I wonder if it wouldn't be better to use e.g. `LRV1` and `LRLatest` names. Are we confident that there's no chance for `LiquidityReceiverV3` in the future?

File: scripts/networks/deploymentFunctions.ts [hunk 4]
```diff
@@ -1833,3 +1835,45 @@ export const prechecks: IDeploymentFunction = async function (params: IDeploymen
...
+
+  const pyroTokenFactory = await ethers.getContractFactory("PyroToken")
+  const behodlerTokenSection = params.existing[sectionName(Sections.BehodlerTokens)]
+  const behodlerTokenNames = Object.keys(behodlerTokenSection)
+  for (let i = 0; i < behodlerTokenNames.length; i++) {
+    const tokenName = behodlerTokenNames[i] as contractNames
+    const token = await getContract<Types.ERC20>(Sections.BehodlerTokens, tokenName, "ERC20")
+
+    let displayName: string = tokenName
+    let pyroDisplayName = 'pyro(' + tokenName + ')'
```
Comment (khazra) [inline scripts/networks/deploymentFunctions.ts:1853]: Any particular reason you prefer using string concatenation over template literals? I find `pyro(${tokenName})` to be much more readable than `'pyro(' + tokenName + ')'`, but maybe that's just me.

File: scripts/networks/orchestrate.ts [hunk 5]
```diff
@@ -131,6 +142,26 @@ class Loader {
     return flat;
   }
 
+  async getTokenConfig(recipe: Sections[]): Promise<ITokenConfig[]> {
+
+    const existingKeys = Object.keys(this.existing)
+    const pyroCableSections: Sections[] = [Sections.LiquidityReceiverOld, Sections.LiquidityReceiverNew, Sections.BehodlerTokens]
+
+    const isPyroCapable = pyroCableSections.every(s => recipe.includes(s))
+    console.log('isPyroCapable: ' + isPyroCapable)
```
Comment (khazra) [inline scripts/networks/orchestrate.ts:151]: Maybe replace `console.log` with the defined `logger`?

**Review by khazra — COMMENTED** [review]
Just a few suggestions/questions, nothing major. Overall, this is a great update that should speed up updating dapps by a lot, great work 👏🏻

File: dev-env-app/server/src/api/get-token-configs.ts (hunk 2 above)
Comment (gititGoro) [reply dev-env-app/server/src/api/get-token-configs.ts]: If I recall, this is statically typed as empty set by default but I'll double check the nullability of both those values, thank you.

File: scripts/networks/deploymentFunctions.ts (hunk 3 above)
Comment (gititGoro) [reply scripts/networks/deploymentFunctions.ts:1842]: Ha ha, I left a comment in the UI (still to be committed) telling anyone reading the code to put those questions to bed. It's a weird situation where there are only 2 LRs. What I'm doing in the UI is to refer to LiquidityReceiverV3 as a shorthand for "Liquidity Receiver for PyroTokens V3". The old, new business is confined to the deployment scripts. Nonetheless the issue is contained in the dev-env app and doesn't leak into the broader UI. At some point I think the universal naming scheme of association with the Pyrotokens version is the most sane.
It feels a little like how windows went from 8 to 10, though.

File: scripts/networks/deploymentFunctions.ts (hunk 4 above)
Comment (gititGoro) [reply scripts/networks/deploymentFunctions.ts:1853]: I almost always stick to template literals but in this case, it's swarming with brackets and braces and I find the concatenation approach more readable at a glance.

File: scripts/networks/orchestrate.ts (hunk 5 above)
Comment (gititGoro) [reply scripts/networks/orchestrate.ts:151]: logger is for logical debugging whereas console.log means I accidentally didn't delete all the console.logs before committing.

**Pull request merged**